File: core/src/main/java/uk/co/amlcurran/social/EmptyCalendarItem.java
package uk.co.amlcurran.social;

public class EmptyCalendarItem implements CalendarItem {

    private final Timestamp startTime;
    private final Timestamp endTime;

    public EmptyCalendarItem(Timestamp startTime, Timestamp endTime) {
        this.startTime = startTime;
        this.endTime = endTime;
    }

    @Override
    public String title() {
        return "Empty";
    }

    @Override
    public Timestamp startTime() {
        return startTime;
    }

    @Override
    public Timestamp endTime() {
        return endTime;
    }

    @Override
    public boolean isEmpty() {
        return true;
    }
}

File: android/shared/src/commonMain/kotlin/uk/co/amlcurran/social/EmptyCalendarItem.kt
package uk.co.amlcurran.social

import kotlinx.datetime.Instant

class EmptyCalendarItem(
    override val startTime: Instant,
    override val endTime: Instant
) :
    CalendarItem {

    override val title: String
        get() {
            return "Empty"
        }

    override var isEmpty: Boolean = true
}
